tests/0800-fuzzy.c

```diff
@@ -0,0 +1,195 @@
+      if (error_code == RD_KAFKA_RESP_ERR_NO_ERROR) {
+        TEST_SAY("produce commit ok: %s\n", msg);
+      } else {
+        TEST_SAY("produce commit error: %s, code: %d\n", msg, error_code);
+        i--;
+      }
+    }
+    /* Destroy topic */
+    rd_kafka_topic_destroy(rkt);
+
+    /* Destroy rdkafka instance */
+    TEST_SAY("Destroying kafka instance %s\n", rd_kafka_name(rk));
+    rd_kafka_destroy(rk);
+    return NULL;
+}
+
+void* Consumer(void* arg) {
+    const char* topic_name  = NULL;
+
+    rd_kafka_t* rk;
+    rd_kafka_topic_t* rkt;
+    rd_kafka_conf_t* conf;
+    rd_kafka_topic_conf_t* topic_conf;
+
+    test_conf_init(&conf, &topic_conf, 2 * timeout);
+    rd_kafka_conf_set_error_cb(conf, NULL);
+    if (!topic_name) {
+        topic_name = test_mk_topic_name(k_topic_name, 0);
+    }
+    rk = test_create_handle(RD_KAFKA_CONSUMER, conf);
+
+    rkt = rd_kafka_topic_new(rk, topic_name, topic_conf);
+    if (!rkt) {
+        TEST_SAY("consume thread error\n");
+        return NULL;
+    }
+
+    rd_kafka_consume_start(rkt, 0, RD_KAFKA_OFFSET_BEGINNING);
+    int retry_index = 0;
+    while (1) {
+      rd_kafka_message_t *rkm = rd_kafka_consume(rkt, 0, 1000);
+      if (!rkm) {
+        if (retry_index++ > 80) {
+          break;
+        }
+        TEST_SAY("retry consume: %d\n", retry_index);
+        continue; /* Timeout: no message within 1000ms, retry again. */
+      }
+      retry_index = 0;
+      if (rkm->err) {
+        /* Consumer errors are generally to be considered
+         * informational as the consumer will automatically
+         * try to recover from all types of errors. */
+        TEST_SAY("%% consume error: %s\n", rd_kafka_message_errstr(rkm));
+        rd_kafka_message_destroy(rkm);
+        continue;
+      }
+      if (rkm->payload) {
+        char buf[128];
+        rd_snprintf(buf, sizeof(buf), "%.*s", (int)rkm->len,
+                    (char *)rkm->payload);
+        TEST_SAY("consume value ok: %s\n", buf);
+      }
+      rd_kafka_message_destroy(rkm);
+    }
+    rd_kafka_consume_stop(rkt, 0);
+
+    /* Destroy topic */
+    rd_kafka_topic_destroy(rkt);
+
+    /* Destroy rdkafka instance */
+    TEST_SAY("Destroying kafka instance %s\n", rd_kafka_name(rk));
+    rd_kafka_destroy(rk);
+
+    return NULL;
+}
+
+void* Fuzzy(void* arg) {
+    TEST_SAY("Fuzzy thread start");
+    sleep(5);
+    const char* stop_cmd = "sh single_kafka_controller.sh stop_only 0";
+    const char* start_cmd = "sh single_kafka_controller.sh start_only 0";
+    int index = 0;
+    while (++index < 50) {
+        TEST_SAY("stop kafka, retry: %d\n", index);
+        system(stop_cmd);
+        sleep(3);
+        TEST_SAY("start kafka, retry: %d\n", index);
+        system(start_cmd);
+        sleep(6);
+    }
+    TEST_SAY("Last start kafka, make sure kafka is running\n");
+    system(start_cmd);
+    TEST_SAY("Fuzzy thread finish");
+    return NULL;
+}
+
+int main_0800_fuzzy(int argc, char** argv) {
```

Comment:
The test case is an integration test, which code path should integration tests be?